app/assets/javascripts/discourse/routes/unknown.js.es6

```diff
@@ -1,8 +1,27 @@
 import DiscourseRoute from "discourse/routes/discourse";
+import RSVP from "rsvp";
 import { ajax } from "discourse/lib/ajax";
 
 export default DiscourseRoute.extend({
-  model() {
-    return ajax("/404-body", { dataType: "html" });
+  model(params) {
+    const bodyReq = ajax("/404-body", { dataType: "html" });
+    const testReq = ajax("/permalink-check.json", {
+      data: { path: params.path }
+    });
+    return RSVP.allSettled([bodyReq, testReq]).then(results => {
+      if (results[1].state === "fulfilled" && results[1].value.found) {
```

Comment:
I'm unclear what's the purpose of found given it will raise 404 server side if not found?